# Week 3

<hr>

## Question 1 

The weights of steers in a herd are distributed normally.  The variance is 40,000 and the mean steer weight is 1300 lbs.  Find the probability that the weight of a randomly selected steer is greater than 979 lbs. (Round your answer to 4 decimal places)


In [42]:
import sympy as sym
import numpy as np
import math
sym.init_printing()
from IPython.display import display,Math
import statistics

x = sym.symbols("x")
mu = sym.symbols("mu") 
sigma = sym.symbols("sigma") 

variance = 40000
mean = 1300
std = math.sqrt(variance) 

# convert to z, solve via table.
z = (x - mu) / sigma
display(Math("z = %s" %(sym.latex(z))))

z = (979 - 1300) / std
print(f"Z Score = {z}")

Z = sym.sympify(np.array(
[['z',  .00, .01, .02, .03, .04, .05, .06, .07, .08, .09 ],
 [-1.7, .04457, .04363,.04272,.04182,.04093,.04006,.03920,.03836,.03754,.03673],
 [-1.6,.05480,.05370,.05262,.05155,.05050,.04947,.04846,.04746,.04648,.04551],
 [-1.5,.06681,.06552,.06426,.06301,.06178,.06057,.05938,.05821,.05705,.05592]
]))

display(Math("Z Scores :  %s" %(sym.latex(Z))))
print("one tail probability of 979 given  1300 is 0.495%") 


<IPython.core.display.Math object>

Z Score = -1.605


<IPython.core.display.Math object>

one tail probability of 979 given  1300 is 0.495%


<hr>

## Question Two 
SVGA monitors manufactured by TSI Electronics have life spans that have a normal distribution with a variance of 1,960,000 and a mean life span of 11,000 hours.  If a SVGA monitor is selected at random, find the probability that the life span of the monitor will be more than 8340 hours. (Round your answer to 4 decimal places)

In [104]:
x = sym.symbols("x")
mu = sym.symbols("mu") 
sigma = sym.symbols("sigma") 

variance = 1960000
mean = 11000
std = math.sqrt(variance)  ## 1400 

# convert to z, solve via table.
z = (x - mu) / sigma
display(Math("z = %s" %(sym.latex(z))))
z = (8340 - 11000) / std
print(f"Z Score = {z}")
probability = 1 - stats.norm.cdf(z) 
print()
print(f"Probability {round(probability,4)}") 


<IPython.core.display.Math object>

Z Score = -1.9

Probability 0.9713


<hr>

## Question 3

Suppose the mean income of firms in the industry for a year is 80 million dollars with a standard deviation of 3 million dollars.  If incomes for the industry are distributed normally, what is the probability that a randomly selected firm will earn between 83 and 85 million dollars? (Round your answer to 4 decimal places)

In [187]:
# Using this:  hportal.org/calculators/statistics-calculator/normal-distribution-calculator.php
print("p(83M<X<85M) = 0.1112  (from a website calculator)") 

#### or using scipy.stats: 
import scipy.stats as stats
# mean and standard deviation
mu = 80
sigma = 3
# calculate the cumulative distribution function at 83 and 85
cdf_83 = stats.norm.cdf(83, mu, sigma)
cdf_85 = stats.norm.cdf(85, mu, sigma)

# the probability of a firm earning between 83 and 85 million is the difference of these two
probability = cdf_85 - cdf_83

# print the probability, rounding to 4 decimal places
print(f"\np(83M<X<85M) = {round(probability, 4)}  (from this jupyter block)")


p(83M<X<85M) = 0.1112  (from a website calculator)

p(83M<X<85M) = 0.1109  (from this jupyter block)


<hr>

## Question 4


Suppose GRE Verbal scores are normally distributed with a mean of 456 and a standard deviation of 123.  A university plans to offer tutoring jobs to students whose scores are in the top 14%.  What is the minimum score required for the job offer?  Round your answer to the nearest whole number, if necessary.

In [188]:
# the hard part here is knowing how/why to look up what number in the Z table.
# https://www.inchcalculator.com/z-table/

# 1. Go lookup the Z score that corresponds to postiive 14% / 86%.
Z = 1.08

# 2. Now plug that into the formula for Z Scores, and solove.
z = (x - mu) / sigma
# 3. Now move everything over to the other side and solve for x.

display(Math("1.08 = %s = 1.08 * \\sigma + \\mu =1.08 * 123 +  456 " %(sym.latex(z))))
display(Math("\\text{GRE Score must be > 589}"))
      




<IPython.core.display.Math object>

<IPython.core.display.Math object>

<hr>


## Question 5

The lengths of nails produced in a factory are normally distributed with a mean of 6.13 centimeters and a standard deviation of 0.06 centimeters.  Find the two lengths that separate the top 7% and the bottom 7%.  These lengths could serve as limits used to identify which nails should be rejected.  Round your answer to the nearest hundredth, if necessary.


In [105]:
## like above, find top and bottom 7% from z score table. 
## use stats.norm percent points function

import scipy.stats as stats

# mean and standard deviation
mu = 6.13
sigma = 0.06

# Use the percent point function (inverse of the CDF) to find the 
# lower and upper limits corresponding to the bottom 7% and top 7% 
lower_limit = stats.norm.ppf(0.07, mu, sigma)
upper_limit = stats.norm.ppf(0.93, mu, sigma)

# Round the limits to two decimal places
lower_limit = round(lower_limit, 2)
upper_limit = round(upper_limit, 2)

print("Lower limit:", lower_limit, "cm")
print("Upper limit:", upper_limit, "cm")

Lower limit: 6.04 cm
Upper limit: 6.22 cm


<hr>

## Question 6

An English professor assigns letter grades on a test according to the following scheme.
* A:  Top 13% of scores
* B:  Scores below the top 13% and above the bottom 55%
* C:  Scores below the top 45% and above the bottom 20%
* D:  Scores below the top 80% and above the bottom 9%
* F:  Bottom 9% of scores

Scores on the test are normally distributed with a mean of 78.8 and a standard deviation of 9.8.  Find the numerical limits for a C grade.  Round your answers to the nearest whole number, if necessary.

In [128]:
# almost identical to question 5, find numerical boundaries of 45 and 20.

import scipy.stats as stats

mu = 78.8
sigma = 9.8

c_lower_limit = round(stats.norm.ppf(0.20, mu, sigma))
c_upper_limit = round(stats.norm.ppf(0.45, mu, sigma)) 


print(f"Grade C bands -> {c_lower_limit} {c_upper_limit}") 

Grade C bands -> 71 78


<hr>

## Question 7

Suppose ACT Composite scores are normally distributed with a mean of 21.2 and a standard deviation of 5.4.  A university plans to admit students whose scores are in the top 45%.  What is the minimum score required for admission?  Round your answer to the nearest tenth, if necessary.


In [137]:
import scipy.stats as stats

mu = 21.2
sigma = 5.4

lower_limit = round(stats.norm.ppf(1-.45, mu, sigma), 1)
print(f"ACT lower limit admission score: {lower_limit}") 

ACT lower limit admission score: 21.9


<hr>

## Question 8

Consider the probability that less than 11 out of 151 students will not graduate on time.  Assume the probability that a given student will not graduate on time is 9%.  Approximate the probability using the normal distribution. (Round your answer to 4 decimal places.)


In [164]:
import scipy.stats as stats

## bernoulli trial / binomial distribution
# μ = np
# σ = sqrt(np(1-p))

n = 151 
p = .09
mu = n * p 
sigma = math.sqrt(n * p * (1-p))

# Value for which we want to find the probability
x = 11.5

# Calculate the Z-score
z_score = (x - mu) / sigma

# Calculate the probability using the CDF
probability = stats.norm.cdf(z_score)

# Round the probability to 4 decimal places
rounded_probability = round(probability, 4)

print(f"Probability that a given student will not graduate on time: {rounded_probability}")

Probability that a given student will not graduate on time: 0.2762


<hr>

## Question 9

The mean lifetime of a tire is 48 months with a standard deviation of 7.  If 147 tires are sampled, what is the probability that the mean of the sample would be greater than 48.83 months? (Round your answer to 4 decimal places)
    

In [180]:
import scipy.stats as stats

# Central Limit Theorem --  when the number of observations n is large, 
# the distribution of the sample mean approximates a normal distribution regardless of 
# the shape of the population distribution.

# The mean (μ) of this distribution is the same as the population mean, and its 
# standard deviation (often called the standard error, σx̄) is the population standard deviation 
# divided by the square root of the sample size:
    
# Population mean and standard error
mu = 48
sigma_x_bar = 7 / math.sqrt(147) 

# Sample mean for which we want to find the probability
x_bar = 48.83

# Calculate the Z-score
z_score = (x_bar - mu) / sigma_x_bar

# Calculate the probability using the survival function (1 - CDF)
probability = round(stats.norm.sf(z_score),4)


print(f"Probability: {probability}")

Probability: 0.0753


<hr>

## Question 10

The quality control manager at a computer manufacturing company believes that the mean life of a computer is 91 months, with a standard deviation of 10.  If he is correct, what is the probability that the mean of a sample of 68 computers would be greater than 93.54 months? (Round your answer to 4 decimal places)
    

In [185]:
import scipy.stats as stats

# Also Central Limit Theorem
mu = 91
sigma_x_bar = 10 / math.sqrt(68) 

# Sample mean for which we want to find the probability
x_bar = 93.54

# Calculate the Z-score
z_score = (x_bar - mu) / sigma_x_bar

# Calculate the probability using the survival function (1 - CDF)
probability = round(stats.norm.sf(z_score),4)

print(f"Probability: {probability}")


Probability: 0.0181


<hr>

## Question 11

A director of reservations believes that 7% of the ticketed passengers are no-shows.  If the director is right, what is the probability that the proportion of no-shows in a sample of 540 ticketed passengers would differ from the population proportion by less than 3%? (Round your answer to 4 decimal places)



In [ ]:
# This is a problem of sampling distributions of proportions. 




<hr>

## Question 12

A bottle maker believes that 23% of his bottles are defective.  If the bottle maker is accurate, what is the probability that the proportion of defective bottles in a sample of 602 bottles would differ from the population proportion by greater than 4%? (Round your answer to 4 decimal places)

In [184]:
# Also a problem related to the sampling distribution of proportions.


<hr>

## Question 13
A research company desires to know the mean consumption of beef per week among males over age 48.  Suppose a sample of size 208 is drawn with x_bar  = 3.9.  Assume sigma = 0.8 .  Construct the 80% confidence interval for the mean number of lb. of beef per week among males over 48. (Round your answers to 1 decimal place)  (answer expects lower and upper bounds) 


<hr>

## Question 14
An economist wants to estimate the mean per capita income (in thousands of dollars) in a major city in California.  Suppose a sample of size 7472 is drawn with x ̅  = 16.6.  Assume ® = 11 .  Construct the 98% confidence interval for the mean per capita income. (Round your answers to 1 decimal place) 
(answer expects lower and upper bounds) 

<hr>

## Question 15 

Find the value of t such that 0.05 of the area under the curve is to the left of t.  Assume the degrees of freedom equals 26.

<hr>

## Question 16

The following measurements ( in picocuries per liter ) were recorded by a set of helium gas detectors installed in a laboratory facility:  383.6, 347.1, 371.9, 347.6, 325.8, 337. Using these measurements, construct a 90% confidence interval for the mean level of helium gas present in the facility.  Assume the population is normally distributed.  

* Step 1. Calculate the sample mean for the given sample data. (Round answer to 2 decimal places) 
* Step 2. Calculate the sample standard deviation for the given sample data. (Round answer to 2 decimal places) 
* Step 3. Find the critical value that should be used in constructing the confidence interval. (Round answer to * 3 decimal places) 
* Step 4. Construct the 90% confidence interval. (Round answer to 2 decimal places)



<hr>

## Question 17

A random sample of 16 fields of spring wheat has a mean yield of 46.4 bushels per acre and standard deviation of 2.45 bushels per acre.  Determine the 80% confidence interval for the true mean yield.  Assume the population is normally distributed.  

* Step 1. Find the critical value that should be used in constructing the confidence interval. (Round answer to 3 decimal places) 
* Step 2. Construct the 80% confidence interval. (Round answer to 1 decimal place)



<hr>

## Question 18

A toy manufacturer wants to know how many new toys children buy each year.  She thinks the mean is 8 toys per year.  Assume a previous study found the standard deviation to be 1.9.  How large of a sample would be required in order to estimate the mean number of toys bought per child at the 99% confidence level with an error of at most 0.13 toys? (Round your answer up to the next integer)


<hr>

## Question 19
A research scientist wants to know how many times per hour a certain strand of bacteria reproduces.  He believes that the mean is 12.6.  Assume the variance is known to be 3.61.  How large of a sample would be required in order to estimate the mean number of reproductions per hour at the 95% confidence level with an error of at most 0.19 reproductions? (Round your answer up to the next integer)


<hr>

## Question 20

The state education commission wants to estimate the fraction of tenth grade students that have reading skills at or below the eighth grade level.

* Step 1. Suppose a sample of 2089 tenth graders is drawn. Of the students sampled, 1734 read above the eighth grade level.  Using the data, estimate the proportion of tenth graders reading at or below the eighth grade level. (Write your answer as a fraction or a decimal number rounded to 3 decimal places) 

* Step 2. Suppose a sample of 2089 tenth graders is drawn.  Of the students sampled, 1734 read above the eighth grade level.  Using the data, construct the 98% confidence interval for the population proportion of tenth graders reading at or below the eighth grade level. (Round your answers to 3 decimal places)





<hr>

## Question 21
An environmentalist wants to find out the fraction of oil tankers that have spills each month.

* Step 1. Suppose a sample of 474 tankers is drawn.  Of these ships, 156 had spills.  Using the data, estimate the proportion of oil tankers that had spills. (Write your answer as a fraction or a decimal number rounded to 3 decimal places) 
* Step 2. Suppose a sample of 474 tankers is drawn.  Of these ships, 156 had spills.  Using the data, construct the 95% confidence interval for the population proportion of oil tankers that have spills each month. (Round your answers to 3 decimal places)





